vscode-lean4/src/utils/projectInfo.ts

```diff
@@ -123,39 +124,20 @@ async function readLeanVersionFile(packageFileUri : Uri) : Promise<string | null
     if (packageFileUri.scheme !== 'file'){
         return null;
     }
+
     if (packageFileUri.path.endsWith(tomlFileName))
     {
-        return await new Promise<string | null>((resolve, reject) => {
-            if (fs.existsSync(url)) {
-                fs.readFile(url, { encoding: 'utf-8' }, (err, data) =>{
-                    if (err) {
-                        reject(err);
-                    } else {
-                        const match = /lean_version\s*=\s*"([^"]*)"/.exec(data.toString());
-                        if (match) resolve(match[1].trim());
-                        resolve(null);
-                    }
-                });
-            } else {
-                resolve(null);
-            }
-        });
+        const data = await fsReadHelper(url);
+        if (data) {
+            const match = /lean_version\s*=\s*"([^"]*)"/.exec(data);
+            if (match) return match[1].trim();
+        }
+
     } else {
         // must be a lean-toolchain file, these are much simpler they only contain a version.
-        return await new Promise<string | null>((resolve, reject) => {
```

Comment:
Nice!  I always love to see lots of red during a code review.  One of the most useful tools in programming is the garbage can :-)

Reply:
Of course :) thanks!